### DISCUSSION TITLE
Integral cumulative error

### DISCUSSION THREAD

**Discussion by doltae09** [bitcraze/discussions#328] [Q&A]
Hello! I have question about integral calculation in "controller_pid.c".

I am designing the State-space equation in "controller_pid.c" below.
![Question_221019](https://user-images.githubusercontent.com/79627957/196581746-1a36cd03-b2b6-41cb-b966-f1205c07cd75.png)

When I check the Crazyflie Client' Plotter, the result as follow:
![image](https://user-images.githubusercontent.com/79627957/196581479-932b8524-ff8b-4043-9b12-422b10170080.png)

In this situation, I expect that the "y_ref" should follow the "rollRateActual", but the result was not.
Is my integral calculation wrong? I want to solve this problem in "controller_pid.c".

**Comment by krichardsson** [comment]
I'm sorry, I don't think I have any input on this. 
We can help you understand the firmware in our repositories and how to interface with it, but when you start to add your own code I'm afraid you will have to debug it yourself.
